File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/agenda/AgendaFragment.java
package com.mentalmachines.droidcon_boston.views.agenda;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.mentalmachines.droidcon_boston.R;
import com.mentalmachines.droidcon_boston.views.base.BaseFragment;

import java.util.Calendar;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 *
 * Created by jinn on 3/11/17.
 */

public class AgendaFragment extends BaseFragment implements AgendaContract.View {

    @BindView(R.id.tablayout)
    android.support.design.widget.TabLayout tabLayout;

    @BindView(R.id.viewpager)
    android.support.v4.view.ViewPager viewPager;

    ScheduleAdapter adapter;


    public static final String TAB_POSITION = "POSITION";

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(getLayout(), container, false);
        ButterKnife.bind(this, rootView);
        return rootView;
    }

    @Override
    public int getLayout() {
        return R.layout.agenda_fragment;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        setupDayPager(view, savedInstanceState);
    }

    @Override
    public void showProgress(boolean progress) {

    }

    @Override
    public void showError(Throwable throwable) {

    }


    private void setupDayPager(View parent, Bundle savedInstanceState) {
        ViewPager viewPager = (ViewPager) parent.findViewById(R.id.viewpager);
        viewPager.setAdapter(new AgendaDayPagerAdapter(getChildFragmentManager()));

        TabLayout tabLayout = (TabLayout) parent.findViewById(R.id.tablayout);
        tabLayout.setupWithViewPager(viewPager);

        if (savedInstanceState != null) {
            viewPager.setCurrentItem(savedInstanceState.getInt(TAB_POSITION));
        } else {
            // set current day to second if today matches
            Calendar today = Calendar.getInstance();
            Calendar dayTwo = Calendar.getInstance();
            dayTwo.set(2017, 4, 10);
            if (today.equals(dayTwo)) {
                viewPager.setCurrentItem(1);
            }
        }
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putInt(TAB_POSITION, tabLayout.getSelectedTabPosition());
    }

    @Override
    public void onViewStateRestored(Bundle savedInstanceState) {
        super.onViewStateRestored(savedInstanceState);
    }
}

File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/agenda/AgendaFragment.kt
package com.mentalmachines.droidcon_boston.views.agenda

import android.os.Bundle
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.mentalmachines.droidcon_boston.BuildConfig
import com.mentalmachines.droidcon_boston.R
import kotlinx.android.synthetic.main.agenda_fragment.*
import java.util.*

class AgendaFragment : Fragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.agenda_fragment, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupDayPager(savedInstanceState)
    }

    private fun setupDayPager(savedInstanceState: Bundle?) {
        viewpager.adapter = AgendaDayPagerAdapter(childFragmentManager, isMyAgenda())

        tablayout.setupWithViewPager(viewpager)

        if (savedInstanceState != null) {
            viewpager.currentItem = savedInstanceState.getInt(TAB_POSITION)
        } else {
            // set current day to second if today matches
            val today = Calendar.getInstance()
            val dayTwo = Calendar.getInstance()
            dayTwo.set(
                BuildConfig.EVENT_YEAR,
                BuildConfig.EVENT_MONTH - 1, // Calendar is 0 indexed
                BuildConfig.EVENT_DAY_TWO
            )
            if (today == dayTwo) {
                viewpager.currentItem = 1
            }
        }
    }

    fun isMyAgenda() = arguments?.getBoolean(ARG_MY_AGENDA) ?: false

    override fun onSaveInstanceState(outState: Bundle) {
        outState.putInt(TAB_POSITION, tablayout.selectedTabPosition)
        super.onSaveInstanceState(outState)
    }

    override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater?) {
        inflater?.inflate(R.menu.menu_search, menu)
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        return when (item?.itemId) {
            R.id.search -> {
                activity?.onSearchRequested()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    companion object {

        const val TAB_POSITION = "POSITION"

        private const val ARG_MY_AGENDA = "my_agenda"

        fun newInstance() = newInstance(false)
        fun newInstanceMySchedule() = newInstance(true)

        private fun newInstance(myAgenda: Boolean): AgendaFragment {
            val fragment = AgendaFragment()
            val args = Bundle()
            args.putBoolean(ARG_MY_AGENDA, myAgenda)
            fragment.arguments = args
            return fragment
        }
    }
}
